base_stage_state/models/base_stage_mixin.py

```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+# © 2016 Daniel Reis
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+
+from openerp import models, fields, api
+
+
+class BaseStageMixin(models.AbstractModel):
+    _name = "base.stage.mixin"
+    _description = "Stage Aware Model"
+
+    @api.model
+    def _get_default_stage(self):
+        """ Gives default stage_id """
+        return self.env['base.stages'].get_default_stage(self._name)
+
+    stage_id = fields.Many2one(
+        'base.stage', 'Stage',
+        track_visibility='always', index=True,
+        domain="[('model_name', '=', self._name)]",
+        default=_get_default_stage,
+        copy=False)
+    state = fields.Selection(
+        related='stage_id.state', readonly=True, store=True)
+
+    # Kanban fields
+    kanban_state = fields.Selection(
+        [('normal', 'Normal'),
+         ('blocked', 'Blocked'),
+         ('done', 'Ready for next stage')],
+        'Kanban State',
+        track_visibility='onchange',
+        default='normal',
+        copy=False,
+        help="The kanban state indicates the workflow readiness:\n"
+             " * Normal is the default situation\n"
+             " * Blocked indicates something is preventing progress\n"
+             " * Ready for next stage indicates it is ready"
+             " to be pulled to the next stage")
+    color = fields.Integer('Color Index')
```

Comment:
Did we maybe want to add in the KanBan legend fields as well? `legend_blocked`, etc.

Also- what about sequencing/priority, or is that outside of the scope of this module?